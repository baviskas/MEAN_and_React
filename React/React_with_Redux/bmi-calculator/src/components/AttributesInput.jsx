import React, { Component } from 'react';

class AttributesInput extends Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
    }

    scaleNames = {
        k: 'kilograms',
        p: 'pounds',
        m: 'metres'
    };

    handleChange(e) {
        this.props.onInputChange(e.target.value);
    }

    render() {
        const scale = this.props.scale;
        return (
            <fieldset>
                <legend>Enter {this.props.attribute} in {this.scaleNames[scale]}:</legend>
                <input value={this.props.value}
                       onChange={this.handleChange} />
            </fieldset>
        );
    }
}

export default AttributesInput;
