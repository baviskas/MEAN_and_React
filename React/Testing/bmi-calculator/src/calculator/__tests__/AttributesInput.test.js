import { shallowWrapper } from "./../../testUtil";
import AttributesInput from "../AttributesInput";

const handler = jest.fn();
const requiredProps = {
    scale: "k",
    attribute: "KiloGram",
    value: 100,
    handler: handler
};
const wrapComponent = (props) => {
    return shallowWrapper(AttributesInput, requiredProps, props);
};

describe("AttributesInput: ", () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    describe("Smoke Test", () => {
        it("render smoketest", () => {
            const wrapper = wrapComponent();
            expect(wrapper.debug()).toMatchSnapshot();
        });
    });

    describe("Event handlers", () => {
        const event = { target: { value: 120 }};
        const wrapper = wrapComponent();
        wrapper.find("input").simulate("change", event);
        expect(handler).toHaveBeenCalled();
    });
});
