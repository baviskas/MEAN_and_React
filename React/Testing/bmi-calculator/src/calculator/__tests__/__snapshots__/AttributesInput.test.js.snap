// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributesInput:  Smoke Test render smoketest 1`] = `
"<fieldset>
  <legend>
    Enter 
    KiloGram
     in 
    Kilograms
  </legend>
  <input value={100} onChange={[Function (anonymous)]} />
</fieldset>"
`;

exports[`VinInput:  Smoke Test render smoketest 1`] = `
"<fieldset>
  <legend>
    Enter 
    KiloGram
     in 
    Kilograms
  </legend>
  <input value={100} onChange={[Function (anonymous)]} />
</fieldset>"
`;
